applications/MeshingApplication/custom_processes/parmmg/pmmg_process.cpp

```diff
@@ -51,10 +56,28 @@ template<PMMGLibrary TPMMGLibrary>
 ParMmgProcess<TPMMGLibrary>::ParMmgProcess(
     ModelPart& rThisModelPart,
     Parameters ThisParameters
-    ):mrThisModelPart(rThisModelPart),
-      mThisParameters(ThisParameters)
+    ): BaseType(&rThisModelPart)
```

Comment:
@philbucher I created a new protected constructor without the parameters to be used by the derived classes

Reply:
@marcnunezc it seems like what you did is a bit different from #7290 
Why didn't you do it exactly in the same way?

not blocking, I think it would be good to address this in a future PR to avoid problems in the future

Reply:
One main difference is that the MmgProcess writes the parameters to a member variable mThisParameters. This made it a bit difficult as only the BaseType can assign in instantation. This is solved by using the extra empty constructor only for derived classes as in #7290 and assigining mThisParameters in the derived constructor. 

Regarding the GetDefaultParameters, I think I am doing the same as #7290.

For the rest (ValidateAndAssignDefaults() and AssignEntities()), overall the main reason is to keep at minium the changes in mmg_process in this PR. 

I agree this can be adressed in a future PR to do exactly as in #7290.

Reply:
ok sounds good!